darts/datasets/__init__.py

```diff
@@ -491,6 +491,32 @@ def __init__(self):
         )
 
 
+class TaxiNewYorkDataset(DatasetLoaderCSV):
```

Comment:
Nit: The name of the class should be equal to the name of the dataset, i.e., `TaxiNewYorkPassengersDataset`.